website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -1325,6 +1351,9 @@ spec:
 ### Custom
 
 * No merging is performed, your UserData must perform all setup required of the node to allow it to join the cluster.
+* Custom UserData must meet the following requirements to work correctly with Karpenter:
+  * It must ensure the node is registered with the `karpenter.sh/unregistered:NoExecute` taint
+  * It must set kubelet config options to match those configured in `spec.kubelet`
```

Comment:
Pretty much, easier to just say do it for all of them though rather than enumerating which fields need to be in both vs just UserData. Just less error prone.